### PR TITLE
API Style Guide information from original document

### PR DESCRIPTION
**Pull request by alicejw-aws**
Signed-off-by: alicejw <alicejw@amazon.com>

### Description
The original API Style Guide template quip that I provided contained more detailed instructions. 
I've added them to this pull request. 

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: STYLE_API_TEMPLATE.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# API reference page template
+# API reference style guide
 
-This template provides the basic structure for creating OpenSearch API documentation. It includes the most important elements that should appear in the documentation and helpful suggestions to help support them. 
+This style guide provides a template with the basic structure for OpenSearch API documentation. It includes the basic headings and helpful suggestions for what information is required in each section.
 
-Depending on the intended purpose of the API, *some sections will be required while others may not be applicable*.
+Depending on the intended purpose of the API, *some sections will be required while others may not be applicable*. For example, only PUT or POST APIs require request and response field descriptions.
```
Comment (Naarcha-AWS) [inline STYLE_API_TEMPLATE.md:5]: Let's remove the second sentence. It's not accurate. See https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/get-index/.

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: STYLE_API_TEMPLATE.md (hunk 1 above)
Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:5]: agreed. thanks!

File: STYLE_API_TEMPLATE.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# API reference page template
+# API reference style guide
```
Comment (Naarcha-AWS) [inline STYLE_API_TEMPLATE.md:1]: This page was never meant to be a style guide, rather, a template that any external contributor could use to write their own API documentation. Please change this back.

File: STYLE_API_TEMPLATE.md [hunk 3]
```diff
@@ -29,31 +29,34 @@ The following sections describe the basic API documentation structure. Each sect
...
+1. (Optional) Path parameter usage examples (heading level 3)
 1. Path parameters (heading level 3)
 1. Query parameters (heading level 3)
 1. Request fields (heading level 3)
 1. Sample request (heading level 4)
 1. Sample response (heading level 4)
 1. Response fields (heading level 3)
 
-## API name
+## {Name of API} API
```
Comment (Naarcha-AWS) [inline STYLE_API_TEMPLATE.md:40]: According to the OpenAPI initiative, an HTTP path is the CRUD operations before the endpoint, such as GET, PUT, and Delete. With that in mind, we do need to include the HTTP methods for each API, which is why this header was named "Path and HTTP methods". 

For reference, see the OpenAPI documentation: https://swagger.io/docs/specification/paths-and-operations/

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: STYLE_API_TEMPLATE.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# API reference page template
+# API reference style guide
 
-This template provides the basic structure for creating OpenSearch API documentation. It includes the most important elements that should appear in the documentation and helpful suggestions to help support them. 
+This style guide provides a template with the basic structure for OpenSearch API documentation. It includes the basic headings and helpful suggestions for what information is required in each section.
```
Comment (Naarcha-AWS) [inline STYLE_API_TEMPLATE.md:3]: This isn't a style guide, just a template.

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: STYLE_API_TEMPLATE.md (hunk 4 above)
Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:3]: actually, a template would contain only pertinent sections with specific examples for each. We have provided a style guide with guidelines for how to write those sections.

File: STYLE_API_TEMPLATE.md (hunk 2 above)
Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:1]: Correct. The original template I provided with sections and descriptions was an actual template for external contributors. 

However, after several rewrites this page contains style guidelines now. 
Hence, it should be titled appropriately as to what it currently provides.

File: STYLE_API_TEMPLATE.md (hunk 3 above)
Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:40]: Yes. That is correct, however the definition of HTTP methods (GET, PUT, POST, etc.) is not in question.

There are several inaccuracies with the new section title and its contents: 
"Path and HTTP methods" does not describe what we are showing in the section body.
The section contains several API calls with an HTTP method and each of the optional path parameters.

Suggestion to change heading to accurately reflect what is shown in the section: "Path parameter usage."

**Path parameter usage** 

The following examples show various path parameter usage with this API.

```
GET /_nodes
GET /_nodes/<nodeId>
GET /_nodes/<metrics>
GET /_nodes/<nodeId>/<metrics>
GET /_nodes/<nodeId>/info/<metrics>
```

File: STYLE_API_TEMPLATE.md (hunk 4 above)
Comment (hdhalter) [reply STYLE_API_TEMPLATE.md:3]: I think it's ok to have style guidelines in a template. Let's leave it.

File: STYLE_API_TEMPLATE.md (hunk 3 above)
Comment (Naarcha-AWS) [reply STYLE_API_TEMPLATE.md:40]: But not all endpoints have path parameters. All endpoints do have HTTP operations.

Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:40]: correct. then we would not include those in this list of examples, right?

Comment (alicejw-aws) [reply STYLE_API_TEMPLATE.md:40]: just to refocus the issue: 
"Path and HTTP methods" does not describe what we are showing in the section body.
The section contains several API calls with an HTTP method and each of the optional path parameters.

Suggestion to change heading to accurately reflect what is shown in the section: "Path parameter usage."

**Comment by alicejw-aws** [comment]
Closing this PR. There is clear opposition to suggestions to add value here.

**Pull request closed without merging**